istio/oam/register.go

```diff
@@ -117,3 +123,78 @@ func load(basePath string) ([]schemaDefinitionPathSet, error) {
 
 	return res, nil
 }
+
+//RegisterWorkLoadsDynamically ...
+func RegisterWorkLoadsDynamically(runtime, host string) error {
+	release, err := config.GetLatestReleases(1)
+	if err != nil {
+		fmt.Println("Could not get latest stable release")
+		return err
+	}
+	v := release[0].TagName
+	m := manifests.Config{
+		Name:        "Istio",
+		MeshVersion: v,
+		Filter: manifests.CrdFilter{
+			RootFilter:    []string{"$[?(@.kind==\"CustomResourceDefinition\")]"},
+			NameFilter:    []string{"$..[\"spec\"][\"names\"][\"kind\"]"},
+			VersionFilter: []string{"$..spec.versions[0]", " --o-filter", "$[0]"},
+			GroupFilter:   []string{"$..spec", " --o-filter", "$[]"},
+			SpecFilter:    []string{"$..openAPIV3Schema.properties.spec", " --o-filter", "$[]"},
+		},
+	}
+	fmt.Println("version ", v)
+	url := "https://raw.githubusercontent.com/istio/istio/" + v + "/manifests/charts/base/crds/crd-all.gen.yaml"
+	comp, err := manifests.GetFromManifest(url, manifests.SERVICE_MESH, m)
+	if err != nil {
+		return err
+	}
+	for i, def := range comp.Definitions {
+		var ord adapter.OAMRegistrantData
+		ord.OAMRefSchema = comp.Schemas[i]
+
+		//Marshalling the stringified json
+		ord.Host = host
+		definitionMap := map[string]interface{}{}
+		if err := json.Unmarshal([]byte(def), &definitionMap); err != nil {
+			return err
+		}
+		// To be shifted in meshkit
+		definitionMap["apiVersion"] = "core.oam.dev/v1alpha1"
+		definitionMap["kind"] = "WorkloadDefinition"
+		ord.OAMDefinition = definitionMap
+		ord.Metadata = map[string]string{
+			config.OAMAdapterNameMetadataKey: config.IstioOperation,
+		}
+		// send request to the register
+		backoffOpt := backoff.NewExponentialBackOff()
+		backoffOpt.MaxElapsedTime = 10 * time.Minute
```

Comment:
```suggestion
		backoffOpt.MaxElapsedTime = 30 * time.Minute
```
Let’s adjust thes to 30 min timeout - across all adapters please.

Reply:
yups, this whole PR is going to change along with other PR's in other adapters as I have migrated that function in adapter library

Reply:
Does the adapter library have a 30 minute setting?

is the adapter library function complete? If so, best to refactor this adapter prior to merging this PR (better to close this PR, refactor and open a new PR).

Reply:
this change is old. After the function was shifted to adapter library, timeout was configured to be 30 minutes, you can check latest changes.

Reply:
Excellent. 👍